hamilton/src/main.cpp

```diff
@@ -242,62 +246,117 @@ bool executeTrajectory(const trajectory_msgs::JointTrajectory& trajectory)
+
+      // Prepend the first point of the next segment (if the current segment is not the first segment itself) 
+      if(it != unplanned_trajectory.begin())
+      { 
+        std::advance(it, -1);
+        auto last_point_of_previous_segment = it->waypoints_geom_msgs.begin();
+        Eigen::Affine3d last_point_of_previous_segment_eigen;
+        last_point_of_previous_segment_eigen = Eigen::Translation3d(last_point_of_previous_segment->position.x, last_point_of_previous_segment->position.y, last_point_of_previous_segment->position.z);
+        descartes_core::TrajectoryPtPtr last_point_of_previous_segment_descartes = makeTolerancedCartesianPoint(last_point_of_previous_segment_eigen);
+        process_segment_descartes.insert(process_segment_descartes.begin(), last_point_of_previous_segment_descartes);
+        std::advance(it, 1); //let the hybrid iterator come back to the current segment
+      }
+
+      // Call the Descartes planner to do some planning
+      if (!planner.planPath(process_segment_descartes))
+      {
+        ROS_ERROR("Could not solve for a valid path");
+      }
+
+      TrajectoryVec result;
+      if (!planner.getPath(result))
+      {
+        ROS_ERROR("Could not retrieve path");
+      }
+
+      // Get Joint Names
+      std::vector<std::string> names;
+      nh.getParam("controller_joint_names", names);
+      // Generate a ROS joint trajectory with the result path, robot model, given joint names,
+      // a certain time delta between each trajectory point
+      trajectory_msgs::JointTrajectory descartes_joint_solution = toROSJointTrajectory(result, *model, names, 1.0);
+
+      //Append planned path
+      robot_trajectory::RobotTrajectory descartes_robot_traj(group.getCurrentState()->getRobotModel(), "manipulator"); //TODO group.getRobotModel does the same thing. 
+      descartes_robot_traj.setRobotTrajectoryMsg(*group.getCurrentState(), descartes_joint_solution);
+      overall_robot_traj.append(descartes_robot_traj, descartes_robot_traj.getWaypointDurationFromStart(descartes_robot_traj.getWayPointCount())); 
+      ROS_INFO("Appended process segment");
+    }
   }
-
-  // Get Joint Names
-  std::vector<std::string> names;
-  nh.getParam("controller_joint_names", names);
-  // Generate a ROS joint trajectory with the result path, robot model, given joint names,
-  // a certain time delta between each trajectory point
-  trajectory_msgs::JointTrajectory descartes_joint_solution = toROSJointTrajectory(result, *model, names, 1.0);
-
-  //Append planned path
-  robot_trajectory::RobotTrajectory descartes_robot_traj(group.getCurrentState()->getRobotModel(), "manipulator"); //TODO group.getRobotModel does the same thing. 
-  descartes_robot_traj.setRobotTrajectoryMsg(*group.getCurrentState(), descartes_joint_solution);
-  overall_robot_traj.append(descartes_robot_traj, descartes_robot_traj.getWaypointDurationFromStart(descartes_robot_traj.getWayPointCount())); 
-  ROS_INFO("Appended process segment");
 }
+
```

Comment:
Several functions in this main seem to overlap with the methods in the HybridPlanner.  Are some of these functions no longer needed?